trestle/core/models/elements.py

```diff
@@ -58,31 +61,101 @@ def _parse(self, element_path: str) -> List[str]:
+        if use_parent:
+            effective_path = self.get_full_path_parts()
+        else:
+            effective_path = self._path
+
+        if not root_model:
+            # lookup root model from top level oscal models or fail
+            prev_model = self._top_level_type_lookup(effective_path[0])
+        else:
+            prev_model = root_model
+        if len(effective_path) == 1:
+            return prev_model
+        # variables
+        # for current_element_str in effective_path[1:]:
+        for current_element_str in effective_path[1:]:
+            # Determine if the parent model is a collection.
+            if utils.is_collection_field_type(prev_model):
+                inner_model = utils.get_inner_type(prev_model)
+                inner_class_name = utils.classname_to_alias(inner_model.__name__, 'json')
+                # Assert that the current name fits an expected form.
+                # Valid choices here are *, integer (for arrays) and the inner model alias
+                if (inner_class_name == current_element_str or current_element_str == self.WILDCARD
+                        or current_element_str.isnumeric()):
+                    prev_model = inner_model
+
+                else:
+                    raise TrestleError('Unexpected key in element path when finding type.')
+
+            else:
+                # Indices, * are not allowed on non-collection types
+                if current_element_str == self.WILDCARD:
+                    logger.error('Cannot get the type of an element path where wild cards do not match a  ')
+                    raise TrestleError(
+                        'Wild card in unexpected position when trying to find class type.'
+                        + ' Element path type lookup can only occur where a single type can be identified.'
+                    )
+                prev_model = prev_model.alias_to_field_map()[current_element_str].outer_type_
+        return prev_model
+
+    def _top_level_type_lookup(self, element_str: str) -> Type[common_types.TopLevelOscalModel]:
+        """From an individual element tag, induce the type of the model.
+
+        Args:
+            element_tag: individual element as text such as 'catalog' or 'profile'
+
+        Returns:
+            Top level object model such as catalog, profile etc.
+        """
+        # Even though awkward use chain of models.
+        if element_str not in const.MODEL_TYPE_LIST:
+            raise TrestleError(f'{element_str} is not a top level model (e.g. catalog, profile)')
+        model_package = const.MODEL_TYPE_TO_MODEL_MODULE[const.MODEL_TYPE_TO_MODEL_DIR[element_str]]
+        object_type, _ = utils.get_root_model(model_package)
+        object_type = cast(Type[common_types.TopLevelOscalModel], object_type)
+        return object_type
+
+    def is_multipart(self) -> bool:
+        """Assert whether or not an element path is multiple parts.
+
+        Originally element paths had to have multiple paths.
```

Comment:
Please provide examples of what is and is not allowed.

Reply:
Done.